commands/proposal_submission.go

```diff
@@ -918,6 +919,27 @@ func checkNewMarketChangesConfiguration(changes *vegapb.NewMarketConfiguration)
 	errs.Merge(checkSLAParams(changes.LiquiditySlaParameters, "new_market.changes.sla_params"))
 	errs.Merge(checkLiquidityFeeSettings(changes.LiquidityFeeSettings, "new_market.changes.liquidity_fee_settings"))
 	errs.Merge(checkCompositePriceConfiguration(changes.MarkPriceConfiguration, "new_market.changes.mark_price_configuration"))
+	errs.Merge(checkTickSize(changes.TickSize, "new_market.changes"))
+
+	return errs
+}
+
+func checkTickSize(tickSize string, parent string) Errors {
+	errs := NewErrors()
+
+	if len(tickSize) == 0 {
+		errs.AddForProperty(fmt.Sprintf("%s.tick_size", parent), ErrIsRequired)
+		return errs
+	}
+
+	tickSizeU, overflow := num.UintFromString(tickSize, 10)
+	if overflow {
+		errs.AddForProperty(fmt.Sprintf("%s.tick_size", parent), ErrNotAValidInteger)
+	}
+	if tickSizeU.IsZero() || tickSizeU.IsNegative() {
```

Comment:
I think this should be an `else if` so that the invalid integer case doesn't then fall into the must-be-positive case when `tickSizeU` defaults to 0.